ml-agents/mlagents/trainers/common/nn_policy.py

```diff
@@ -199,82 +203,33 @@ def _create_cc_actor(
-            if condition_sigma_on_obs:
-                log_sigma = tf.layers.dense(
-                    hidden_policy,
-                    self.act_size[0],
-                    activation=None,
-                    name="log_sigma",
-                    kernel_initializer=ModelUtils.scaled_init(0.01),
+            if not tanh_squash:
+                distribution = GaussianDistribution(
+                    hidden_policy, self.act_size, reparameterize=reparameterize
                 )
             else:
-                log_sigma = tf.get_variable(
-                    "log_sigma",
-                    [self.act_size[0]],
-                    dtype=tf.float32,
-                    initializer=tf.zeros_initializer(),
+                distribution = TanhSquashedGaussianDistribution(
+                    hidden_policy, self.act_size, reparameterize=reparameterize
                 )
-            log_sigma = tf.clip_by_value(log_sigma, self.log_std_min, self.log_std_max)
-
-            sigma = tf.exp(log_sigma)
-
-            epsilon = tf.random_normal(tf.shape(mu))
-
-            sampled_policy = mu + sigma * epsilon
-
-            # Stop gradient if we're not doing the resampling trick
-            if not reparameterize:
-                sampled_policy_probs = tf.stop_gradient(sampled_policy)
-            else:
-                sampled_policy_probs = sampled_policy
-
-            # Compute probability of model output.
-            _gauss_pre = -0.5 * (
-                ((sampled_policy_probs - mu) / (sigma + EPSILON)) ** 2
-                + 2 * log_sigma
-                + np.log(2 * np.pi)
-            )
-            all_probs = _gauss_pre
-            all_probs = tf.reduce_sum(_gauss_pre, axis=1, keepdims=True)
 
         if tanh_squash:
-            self.output_pre = tf.tanh(sampled_policy)
-
-            # Squash correction
-            all_probs -= tf.reduce_sum(
-                tf.log(1 - self.output_pre ** 2 + EPSILON), axis=1, keepdims=True
-            )
+            self.output_pre = distribution.sample
             self.output = tf.identity(self.output_pre, name="action")
         else:
-            self.output_pre = sampled_policy
+            self.output_pre = distribution.sample
             # Clip and scale output to ensure actions are always within [-1, 1] range.
             output_post = tf.clip_by_value(self.output_pre, -3, 3) / 3
             self.output = tf.identity(output_post, name="action")
 
         self.selected_actions = tf.stop_gradient(self.output)
 
-        self.all_log_probs = tf.identity(all_probs, name="action_probs")
-
-        single_dim_entropy = 0.5 * tf.reduce_mean(
-            tf.log(2 * np.pi * np.e) + 2 * log_sigma
+        self.all_log_probs = tf.identity(
+            distribution.total_log_probs, name="action_probs"
         )
-        # Make entropy the right shape
-        self.entropy = tf.ones_like(tf.reshape(mu[:, 0], [-1])) * single_dim_entropy
+        self.entropy = distribution.entropy
```

Comment:
can we aggregate all occurrences of this line i.e. line 283

Reply:
👍 will do it in a follow-up PR. TODO: broader discussion about standardization of discrete and continuous (requires inference changes on C#)